File: Pyg_shop/interface/src/main/java/top/takefly/core/service/CartService.java
package top.takefly.core.service;

import top.takefly.core.pojo.address.Address;
import top.takefly.core.pojo.entity.Cart;
import top.takefly.core.pojo.user.User;

import java.util.List;

/**
 * 购物车服务
 */
public interface CartService {

    /**
     * 添加商品到购物车列表
     * @param cartList
     * @param itemId
     * @param num
     * @return
     */
    public List<Cart> addGoodsToCartList(List<Cart> cartList,Long itemId,Integer num );

    /**
     * 从redis中读取购物车
     * @param username
     * @return
     */
    public List<Cart> findCartListFromRedis(String username);

    /**
     * 将购物车信息保存到redis中
     * @param username
     * @param cartList
     */
    public void saveCartListToRedis(String username , List<Cart> cartList);

    /**
     * 将两个购物车合并为一个购物车
     * @param cartList1
     * @param cartList2
     * @return
     */
    public List<Cart> mergeCartList(List<Cart> cartList1 , List<Cart> cartList2);

    /**
     * 通过userId来查找用户信息
     * @param name
     * @return
     */
    public List<Address> findUserListByUserId(String name);

    public void add(Address address);
}

File: Pyg_shop/web_portal/src/main/webapp/js/service/CartService.js
app.service('cartService', function ($http) {

    this.findCartList = function () {
        return $http.get('cart/findCartList.do');
    }

    this.addGoodsToCartList = function (itemId, num) {
        return $http.get('cart/addGoodsToCartList.do?itemId=' + itemId + '&num=' + num);
    }

    this.sum = function(cartList){
        var totalValue = {totalMoney: 0, totalNum: 0};
        //var orderItem = {};
        for (var i = 0; i < cartList.length; i++) {
            var cart = cartList[i];
            for (var j = 0; j < cart.orderItemList.length; j++) {
                var orderItem = cart.orderItemList[j];
                //购物车明细
                totalValue.totalNum += orderItem.num;
                totalValue.totalMoney += orderItem.totalFee;
            }
        }

        return totalValue;
    }

    this.findUserListByUserId = function(){
        return $http.get('/cart/findUserListByUserId.do');
    }

    //添加
    this.add = function(entity){
        return $http.post('/cart/add.do' , entity);
    }

    //提交订单
    this.submitOrder = function(order){
        return $http.post("/order/add.do" , order);
    }
});